indico/web/client/js/react/components/Datepicker.jsx

```diff
@@ -0,0 +1,78 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import 'indico/custom_elements/ind_datepicker';
+import {Param, Translate} from 'indico/react/i18n';
+
+export default function Datepicker({format, onChange, value, ...inputProps}) {
+  function handleDateChange(ev) {
+    onChange(ev.target.parentNode.date);
+  }
+
+  return (
+    <ind-date-picker>
+      <input type="text" onChange={handleDateChange} defaultValue={value} {...inputProps} />
+      <button type="button">
+        <Translate as="span">Open a calendar</Translate>
+      </button>
+      <ind-calendar>
+        <dialog>
+          <div className="controls">
+            <button type="button" value="previous-year">
+              <Translate as="span">Previous year</Translate>
+            </button>
+            <button type="button" value="previous-month">
+              <Translate as="span">Previous month</Translate>
+            </button>
+
+            <div className="month-year">
+              <select />
+              <input type="number" min="1900" max="2100" required />
+            </div>
+
+            <button type="button" value="next-month">
+              <Translate as="span">Next month</Translate>
+            </button>
+            <button type="button" value="next-year">
+              <Translate as="span">Next year</Translate>
+            </button>
+          </div>
+
+          <div className="weekdays">
+            <span />
+            <span />
+            <span />
+            <span />
+            <span />
+            <span />
+            <span />
+          </div>
+          <div role="listbox" />
+        </dialog>
+      </ind-calendar>
+
+      <span className="date-format">
+        <Translate>
+          Date format: <Param name="format" value={format} />
```

Comment:
I think this should only be used in the regform field. We don't use this anywhere else in Indico (we use the locale's default date format instead), it's very weird IMHO.

Typically the requested date format is shown as a placeholder in the input field.